### ISSUE TITLE
Parser applies strict mode on files not having the "use strict" flag

### ISSUE THREAD

**Issue by andreas-marschke** [jsdoc3/jsdoc#1210]
I'm working on a codebase supporting older browsers and adding jsdoc to document our code.
We are currently having the problem that the espree parser employed by jsdoc is using `strict mode` even if `"use strict";` is not used in the code. This results in ERRORs being logged during jsdoc runs:

```
>> ERROR: Unable to parse <File>: Line 1532: Delete of an unqualified identifier in strict mode.
>> ERROR: Unable to parse <File>: Line 237: Delete of an unqualified identifier in strict mode.
```

We can not qualify this as it will throw an error in older browsers. Thus we'll keep using the `delete <var>;` line and would rather disable this. 

Looking at espree there is flag to enable/disable `impliedStrictMode` in newer versions but I'm not sure if this also applies here. I'd rather I could disable this in jsdoc if possible.

Thanks in advance!

**Comment by triangle42** [comment]
There is no 'use strict' in my code but, nevertheless, I also get the "Delete of an unqualified identifier in strict mode." error when running JSDoc.

**Comment by andyp1per** [comment]
I see this too. Any progress?

**Comment by jalbam** [comment]
Same problem here. Is there any way to disable the default implied strict mode for all files (or at least selected ones) parsed by JSDoc easily?

Personally I am getting this:

`ERROR: Unable to parse C:\AppServ\www\prueba_cb\CrossBrowdy\CrossBase\general\CB_Elements.js: Deleting local variable in strict mode
`
CB_Elements.js should not be in strict mode and it should also work for very old browsers.

**Comment by andreas-marschke** [comment]
Hey all! I created a fork for this here: https://github.com/SOASTA/jsdoc , the branch: `fix-no-strict-code-astbuilder` has the change to disable the strict check hard coded. You could use that from instead of mainline jsdoc?

**Comment by jalbam** [comment]
Thank you very much. I wish this feature was available in JSDoc by default, or at least it should be easy to configure.

**Comment by hegemonic** [comment]
Fixed on master. You can now specify a `sourceType` option in your [JSDoc config file](http://usejsdoc.org/about-configuring-jsdoc.html). Valid values are `module` (default) and `script`. If you specify `script`, implied strict mode will be disabled, and you won't be able to use ES2015 modules.

This fix will be included in JSDoc 3.5.0.
